src/OnboardingSPA/pages/Steps/GetStarted/SiteTypeSetup/PrimarySite/index.js

```diff
@@ -94,63 +95,65 @@ const StepPrimarySetup = () => {
-								className={ `${
-									clickedIndex === idx ||
-									item.name === selectedPrimaryCategoryInStore
-										? 'chosenPrimaryCategory '
-										: ''
-								}nfd-card-category` }
-								onClick={ ( e ) => handleCategoryClick( idx ) }
-							>
-								<div className="nfd-card-category-wrapper">
-									<span
-										className="icon"
-										style={ {
-											backgroundImage:
-												clickedIndex !== idx &&
-												item.name !==
-													selectedPrimaryCategoryInStore
-													? item?.icon
-													: item?.iconWhite,
-										} }
-									></span>
-									<span className="categName">
-										{ item?.name }
-									</span>
+				<StepStateHandler watch={ content.categories }>
+					<div className="nfd-setup-primary-categories">
+						{ content.categories.map( ( item, idx ) => {
+							return (
+								<div
+									key={ item?.name }
+									className={ `${
+										clickedIndex === idx ||
+										item.name === selectedPrimaryCategoryInStore
+											? 'chosenPrimaryCategory '
+											: ''
+									}nfd-card-category` }
+									onClick={ ( e ) => handleCategoryClick( idx ) }
+								>
+									<div className="nfd-card-category-wrapper">
+										<span
+											className="icon"
+											style={ {
+												backgroundImage:
+													clickedIndex !== idx &&
+													item.name !==
+														selectedPrimaryCategoryInStore
+														? item?.icon
+														: item?.iconWhite,
+											} }
+										></span>
+										<span className="categName">
+											{ item?.name }
+										</span>
+									</div>
 								</div>
-							</div>
-						);
-					} ) }
-				</div>
-
-				<div className="nfd-setup-primary-second">
-					<div className="nfd-setup-primary-second-top">
-						<p className="blackText">or tell us here:</p>
-						<input
-							type="text"
-							onChange={ ( e ) => categoryInput( e ) }
-							className="tellUsInput"
-							placeholder={ sprintf(
-								__(
-									content.placeholderSiteTypeInput,
-									'wp-module-onboarding'
-								),
-								translations( 'site' )
-							) }
-							value={ inputCategVal }
-						/>
+							);
+						} ) }
 					</div>
-					<div className="nfd-setup-primary-second-bottom">
-						<NavCardButton text={ __( content.buttonText ) } />
-						<NeedHelpTag />
+
+					<div className="nfd-setup-primary-second">
+						<div className="nfd-setup-primary-second-top">
+							<p className="blackText">or tell us here:</p>
+							<input
+								type="text"
+								onChange={ ( e ) => categoryInput( e ) }
+								className="tellUsInput"
+								placeholder={ sprintf(
+									__(
+										content.placeholderSiteTypeInput,
+										'wp-module-onboarding'
+									),
+									translations( 'site' )
+								) }
+								value={ inputCategVal }
+							/>
+						</div>
 					</div>
-				</div>
+				</StepStateHandler>
+				<NavCardButton
```

Comment:
Can we add the NavCardButton and NedHelpTag in the StepStateHandler Tag so that user does not go to the next step if this page takes time to load.
![image](https://user-images.githubusercontent.com/53893563/211769976-06539b5e-e5f0-452f-a96e-d0bd1ee7371d.png)

Reply:
not needed for now as the user can still navigate away using the next and back buttons or the browser buttons